src/common/util/useCurrentPerson.ts

```diff
@@ -0,0 +1,33 @@
+import { useKeycloak } from '@react-keycloak/ssr'
+import { AxiosResponse } from 'axios'
+import { Person, UpdatePerson } from 'gql/person'
+import { KeycloakInstance } from 'keycloak-js'
+import { useQuery } from 'react-query'
+import { apiClient } from 'service/apiClient'
+import { endpoints } from 'service/apiEndpoints'
+import { authConfig, authQueryFnFactory } from 'service/restRequests'
+
+type CurrentPerson = {
+  user: Person
+}
+
+export function useCurrentPerson() {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return useQuery<CurrentPerson>(
+    endpoints.account.me.url,
+    authQueryFnFactory<CurrentPerson>(keycloak?.token),
+  )
+}
+
+export function updateCurrentPerson() {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async (data: UpdatePerson) => {
+    return await apiClient.put<UpdatePerson, AxiosResponse<Person>>(
```

Comment:
Here the response might be:

```suggestion
    return await apiClient.put<UpdatePerson, AxiosResponse<{user: Person}>>(
```